beamer/agent/middleware.py

```diff
@@ -282,3 +282,27 @@ def rate_limiter(
 ) -> Callable[[RPCEndpoint, Any], RPCResponse]:
     state = _RateLimiterState()
     return functools.partial(_rate_limiter, make_request=make_request, w3=w3, state=state)
+
+
+def base_fee_setter(
```

Comment:
I would call this `max_fee_setter` because this is what we are effectively doing